govdelivery/tests/test_api.py

```diff
@@ -1,76 +1,144 @@
+import base64
 import os
 import unittest
-import xml.etree.ElementTree as ET
 
-from govdelivery import api
+import responses
+from requests.models import Response
 
-dummy_username = "foo"
-dummy_password = "bar"
+from govdelivery.api import (
+    authenticated_session, get_full_url_to_call, GovDelivery
+)
+from govdelivery.tests.utils import load_data
+from govdelivery.xml_response_parsers import topic_xml_as_dict
 
 
-def require_env_vars(varnames):
-    for varname in varnames:
-        assert varname in os.environ, "%s must be in os.environ" % varname
+class TestClientUtilities(unittest.TestCase):
+    def test_get_full_url_to_call(self):
+        # Store and unset existing GOVDELIVERY_BASE_URL env var, if it is set.
```

Comment:
This code is pretty complex, and also not thread-safe (if that was ever something to care about in future). Would it be simpler to just `patch` the call to `os.getenv`, to avoid the need to store/restore the variable?

Reply:
Yeah, that might be simpler. Could you show me how to do that? :)

Reply:
How about something like [this](https://gist.github.com/chosak/25bbb54b7181223206ef43b9be1d90d0)?

Reply:
Thanks, implemented in 88ee0a6.